### PR TITLE
Add new CI Actions for JavaScript NextGen

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Add new CI Actions for JavaScript NextGen

**Why**:

* Reduce maintenance!!!

**How**:

**Checklist**:

- [ ] Tests
- [ ] CHANGELOG.md entry
- [ ] Documentation

### CODE REVIEW

File: install-javascript/action.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      env:
+        cache-name: cache-node-modules
+      with:
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
```
Comment (bjoernricks) [inline install-javascript/action.yaml:33]: I am not sure if this is fine in all cases but imho it's up to the JS devs to work on this.

File: lint-javascript/action.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    description: "Token with GitHub Packages read rights required to install."
+    required: true
+branding:
+  icon: "package"
+  color: "green"
+runs:
+  using: "composite"
+  steps:
+    - run: echo "Install project"
+      shell: bash
```
Comment (bjoernricks) [inline lint-javascript/action.yaml:19]:
```suggestion
```

File: lint-javascript/action.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Install Project
+      uses: greenbone/actions/install-javascript@v2
+      with:
+        version: ${{ inputs.version }}
+        token: ${{ inputs.token }}
+    - name: Run linting
+      run: npm run lint
+      shell: bash
```
Comment (bjoernricks) [inline lint-javascript/action.yaml:27]:
```suggestion
```

File: lint-javascript/action.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Run formatting
+      run: npm run pretty:check
+      shell: bash
```
Comment (bjoernricks) [inline lint-javascript/action.yaml:30]: I don't thin setting the shell is necessary here
```suggestion
```

File: test-javascript/action.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    description: "Token with GitHub Packages read rights required to install."
+    required: true
+branding:
+  icon: "package"
+  color: "green"
+runs:
+  using: "composite"
+  steps:
+    - run: echo "Install project"
+      shell: bash
```
Comment (bjoernricks) [inline test-javascript/action.yaml:19]:
```suggestion
```

File: test-javascript/action.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Install Project
+      uses: greenbone/actions/install-javascript@v2
+      with:
+        version: ${{ inputs.version }}
+        token: ${{ inputs.token }}
+    - name: Build ...
+      run: npm run build
+      shell: bash
```
Comment (bjoernricks) [inline test-javascript/action.yaml:27]:
```suggestion
```

File: test-javascript/action.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Run Testing
+      run: npm run test:ci
+      shell: bash
```
Comment (bjoernricks) [inline test-javascript/action.yaml:30]:
```suggestion
```

File: install-javascript/action.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    required: true
+branding:
+  icon: "package"
+  color: "green"
+runs:
+  using: "composite"
+  steps:
+    - run: echo "Checkout Repository"
+      shell: bash
+    - uses: actions/checkout@v3
```
Comment (bjoernricks) [inline install-javascript/action.yaml:19]:
```suggestion
    - name: Checkout Repository
      uses: actions/checkout@v3
```

File: install-javascript/action.yaml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    - name: Cache node modules
+      shell: bash
```
Comment (bjoernricks) [inline install-javascript/action.yaml:21]:
```suggestion
```

File: install-javascript/action.yaml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        # npm cache files are stored in `~/.npm` on Linux/macOS
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-build-${{ env.cache-name }}-
+          ${{ runner.os }}-build-
+          ${{ runner.os }}-
+    - if: ${{ steps.cache-npm.outputs.cache-hit == 'false' }}
+      name: List the state of node modules
+      shell: bash
```
Comment (bjoernricks) [inline install-javascript/action.yaml:36]:
```suggestion
```

File: install-javascript/action.yaml [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+      continue-on-error: true
+      run: npm list
+    - name: Use Node.js ${{ inputs.version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.version }}
+        registry-url: 'https://npm.pkg.github.com'
+    - run: npm ci
+      shell: bash
```
Comment (bjoernricks) [inline install-javascript/action.yaml:45]:
```suggestion
```

File: install-javascript/action.yaml (hunk 1 above)
Comment (y0urself) [reply install-javascript/action.yaml:33]: So we should create an input variable for the `key`?

File: install-javascript/action.yaml (hunk 8 above)
Comment (bjoernricks) [reply install-javascript/action.yaml:19]: Not sure if something should be printed to the console.

**Review by bjoernricks — CHANGES_REQUESTED** [review]
I don't thing the shell needs to be set.

**Comment by y0urself** [comment]
> I don't thing the shell needs to be set.

according to https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file the `shell` needs to be set for `composite` actions?

File: install-javascript/action.yaml (hunk 1 above)
Comment (bjoernricks) [reply install-javascript/action.yaml:33]: I remember that I setting the cache key was not that obvious for JS and yarn with CircleCI in the past. Maybe it is better to use an input variable.

**Comment by bjoernricks** [comment]
> > I don't thing the shell needs to be set.
> 
> according to https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file the `shell` needs to be set for `composite` actions?

Sorry yes indeed it is required for actions.

File: install-npm/action.yaml [hunk 12]
```diff
@@ -0,0 +1,48 @@
+name: "JavaScript Installer [npm]"
+author: "Jaspar Stach <jaspar.stach@greenbone.net>"
+description: "An action to install Greenbone JavaScript Projects"
+inputs:
+  version:
+    description: "Node version that should be installed"
+    default: "16.x"
+  token:
+    description: "Token with GitHub Packages read rights required to install."
+    required: true
```
Comment (bjoernricks) [inline install-npm/action.yaml:10]: Maybe add the default?
```suggestion
    required: true
    default: ${{ github.token }}
```

File: lint-npm/action.yaml [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+name: "JavaScript Linting [npm]"
+author: "Jaspar Stach <jaspar.stach@greenbone.net>"
+description: "An action to check linting and formatting Greenbone JavaScript Projects"
+inputs:
+  version:
+    description: "Node version that should be installed"
+    default: "16.x"
+  token:
+    description: "Token with GitHub Packages read rights required to install."
+    required: true
```
Comment (bjoernricks) [inline lint-npm/action.yaml:11]:
```suggestion
    required: true
    default: ${{ github.token }}
```

File: test-npm/action.yaml [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+name: "JavaScript Build & Test [npm]"
+author: "Jaspar Stach <jaspar.stach@greenbone.net>"
+description: "An action to build and test Greenbone JavaScript Projects [npm]"
+inputs:
+  version:
+    description: "Node version that should be installed"
+    default: "16.x"
+  token:
+    description: "Token with GitHub Packages read rights required to install."
+    required: true
```
Comment (bjoernricks) [inline test-npm/action.yaml:11]:
```suggestion
    required: true
    default: ${{ github.token }}
```

**Review by bjoernricks — APPROVED** [review]

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**